src/Features/Core/Portable/MakeMethodStatic/AbstractMakeMethodStaticDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.MakeMethodStatic
+{
+    internal abstract class AbstractMakeMethodStaticDiagnosticAnalyzer<TSyntaxKind, TMethodDeclarationSyntax>
+        : AbstractCodeStyleDiagnosticAnalyzer
+        where TSyntaxKind : struct
+        where TMethodDeclarationSyntax : SyntaxNode
+    {
+        protected AbstractMakeMethodStaticDiagnosticAnalyzer(LocalizableString title, LocalizableString message)
+            : base(IDEDiagnosticIds.MakeMethodStaticDiagnosticId, title, message)
+        {
+        }
+
+        protected sealed override void InitializeWorker(AnalysisContext context)
+            => context.RegisterSyntaxNodeAction(AnalyzeNode, GetMethodDeclarationSyntaxKind());
+
+        public sealed override bool OpenFileOnly(Workspace workspace) => true;
+        public sealed override DiagnosticAnalyzerCategory GetAnalyzerCategory()
+            => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        private void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var declaration = (TMethodDeclarationSyntax)context.Node;
```

Comment:
This feature needs to be provided with a user option.  Some people are not going to like having all these warnings show up in their code, and we'll need a way for them to disable this.